test/python/circuit/test_controlled_gate.py

```diff
@@ -700,11 +700,17 @@ def test_base_gate_setting(self):
         Test all gates in standard extensions which are of type ControlledGate and have a base gate
         setting.
         """
+        from qiskit.extensions.standard import MCU1Gate
         params = [0.1 * i for i in range(10)]
+        num_ctrl_qubits = 5
         for gate_class in ControlledGate.__subclasses__():
             sig = signature(gate_class.__init__)
             free_params = len(sig.parameters) - 1  # subtract "self"
-            base_gate = gate_class(*params[0:free_params])
+            if gate_class == MCU1Gate:
+                # the multi-controlled gates also take the number of control qubits
+                base_gate = gate_class(params[0], num_ctrl_qubits)
+            else:
+                base_gate = gate_class(*params[0:free_params])
```

Comment:
As far as I'm aware, MCU1 is the only multi-controlled gate that exists right now. I started to implement the `MCXGate` (and after that the multi-controlled rotations) though and I'll add them there.

Reply:
Perhaps `num_ctrl_qubits` could be a keyword argument like it is for the class.

Reply:
Do you mean making `num_ctrl_qubits` a keyword argument in `MCU1Gate`? Because currently it is a positional argument. I changed the above lines of code to 
```python
free_params = params[0:len(sig.parameters) - 1]  # subtract "self"
if gate_class in [MCU1Gate]:  # last argument is the number of control qubits
    free_params[-1] = num_ctrl_qubits

base_gate = gate_class(*free_params)
```
which is now also easily generalized for `MCX` and the `MC`-rotations.

Reply:
I think as this test points out it would be sensible if all members of a gate class had the same call signature. One way to perhaps resolve this would be to use a keyword argument. Another would be to define a new class.

If possible I think the prior is preferable since I'm not sure if there is a transpiler pass which cares about distinguishing e.g. cu1 from mcu1. This also speaks to your first question in the pr description; if these are eventually merged it would be good to have them share a signature now.

Reply:
This is outdated, since I moved `MCU1Gate` into `CU1Gate` 🙂 Now these special distinctions in the tests are  not needed anymore.